src/EnergyPlus/WaterUse.cc

```diff
@@ -603,6 +603,15 @@ namespace WaterUse {
                     state, waterConnection.InletNode, waterConnection.PeakMassFlowRate / Psychrometrics::RhoH2O(DataGlobalConstants::InitConvTemp));
             }
         }
+        // need a good place to set a bool to calculate WaterUse hot and cold flow rates in CalcEquipmentFlowRates
+        // WaterUse can be used with or without WaterUse:Connections, with or without WaterUse:Equipment hot temp schedule
+        for (int WaterEquipNum = 1; WaterEquipNum <= state.dataWaterUse->numWaterEquipment; ++WaterEquipNum) {
+            // set logical if either hot water temp or target temp schedule are missing (will use cold water otherwise)
+            // if a connections object is used then don't need to hot temp schedule
+            state.dataWaterUse->WaterEquipment(WaterEquipNum).allowHotControl =
+                state.dataWaterUse->WaterEquipment(WaterEquipNum).TargetTempSchedule &&
+                (state.dataWaterUse->WaterEquipment(WaterEquipNum).HotTempSchedule || state.dataWaterUse->WaterEquipment(WaterEquipNum).Connections);
+        }
```

Comment:
Can use `auto &thisWaterEquip` here for readability. Actually, if you are not even using the `WaterEquipNum` index for anything else, you can use a range loop (an @mjwitte favorite!) `for (auto &thisWaterEquip : state.dataWaterUse->WaterEquipment)`.